src/lib.rs

```diff
@@ -2533,6 +2533,21 @@ impl Build {
                 }
                 None => default_ar,
             }
+        } else if target.starts_with("wasm32") {
+            // Formally speaking one should be able to use this approach,
+            // parsing -print-search-dirs output, to cover all clang targets,
+            // including Android SDKs and other cross-compilation scenarios...
+            // And even extend it to gcc targets by seaching for "ar" instead
+            // of "llvm-ar"...
+            let compiler = self.get_base_compiler()?;
+            if compiler.family == ToolFamily::Clang {
+                match search_programs(&mut self.cmd(&compiler.path), "llvm-ar") {
+                    Some(ar) => ar.to_str().unwrap().to_owned(),
+                    None => default_ar,
```

Comment:
To avoid hardcoding `default_ar` return values could this be structured with a fall-through to the case below?

Reply:
Well, I seem to fail to figure it out. In order to avoid `default_ar` here one has to do the whole thing in `else if` above. And here is the conundrum. It doesn't let me use `if target.starts_with("wasm32") && let Some(ar) = self.find_ar()`, while `if let Some(ar) = if target.starts_with("wasm32") { self.find_ar() } else { None }` get cargo-fmt-ed in a way that makes it really hard to follow:
```
        } else if let Some(ar) = if target.starts_with("wasm32") {
            self.find_ar()
        } else {
            None
        } {
            ar.to_str().unwrap().to_owned()
        } else {
            default_ar
        };

```